platform/samd11/hal_config.h

```diff
@@ -73,6 +79,10 @@
   #define UART_SERCOM_IRQ_HANDLER  irq_handler_sercom1
   #define UART_SERCOM_TXPO         1
   #define UART_SERCOM_RXPO         3
+
+#elif defined(HAL_BOARD_CUSTOM)
```

Comment:
You can just do #if defined(HAL_BOARD_CUSTOM) as a first check, then you won't need all the code above if this is just to support ability to provide the custom config. And I don't really care  about external configuration for the standard boards.

Reply:
I agree with your suggestion and will update the PR accordingly.